ambry-store/src/main/java/com/github/ambry/store/StoreMetrics.java

```diff
@@ -253,8 +253,12 @@ void initializeIndexGauges(String storeId, final PersistentIndex index, final lo
       Gauge<Long> currentInvalidDataSize = () -> {
         long invalidDataSize = -1;
         try {
-          invalidDataSize = index.getLogUsedCapacity() - blobStoreStats.getValidSize(
-              new TimeRange(System.currentTimeMillis(), blobStoreStats.getBucketSpanTimeInMs())).getSecond();
+          if (storeAsyncGetValidSizeEnable) {
+            invalidDataSize = index.getLogUsedCapacity() - blobStoreStats.getValidSizeAsync().getSecond();
+          } else {
```

Comment:
The else case can be fully removed since it's not workable. 

We can also remove `storeAsyncGetValidSizeEnable`. Once `enableStoreCurrentInvalidSize` is enabled, your solution is the only solution.

Reply:
Currently the non-main clusters are using the previous way, so what I'm planning to do is test on main cluster first, so the logic won't affect other clusters. And after verification, I'll remove the redundant logic. What do you think?

Reply:
Though video and delivery cluster are working, I don't think previous solution is efficient. We will eventually use your solution in this PR. So I suggest to remove the `else` case in this PR directly.